### ISSUE TITLE
formatOnBlur: Also run on submit

### ISSUE THREAD

**Issue by ThiefMaster** [final-form/react-final-form#470]
I'm using `formatOnBlur` to trim strings. This works fine, but if someone submits by pressing the enter key without leaving the field first, then no blur event happens and the string is submitted as-is without the formatter code running.

Or is there a better way to trim all strings before submission? format+formatOnBlur is pretty convenient because we're using a custom Field wrapper component that always sets it for string fields.

**Comment by erikras** [comment]
You could always do the formatting in your submit function before sending it off to the server. Something like:

```js
function onSubmit(values) {
  stringKeys.forEach(key => values[key] = format(values[key]))
  ajax.send(values)
}
```

```jsx
<Form onSubmit={onSubmit}/>
```

**Comment by ThiefMaster** [comment]
I hoped to avoid having to put code for this in every single submit function ;) Also, this wouldn't trigger validation errors for fields that cannot be empty or have a minimum length (which I want to apply on the trimmed value of course).

**Comment by erikras** [comment]
You make good points.

**Comment by erikras** [comment]
You could [use jQuery](https://stackoverflow.com/questions/28124028/how-on-trigger-onblur-before-form-submit)! 🤣 I'm joking...

Because React Final Form is just responding to the events the browser gives it, this is going to require more complex communication between the Final Form engine and the fields.

**Comment by erikras** [comment]
Published fix in [`v5.1.0`](https://github.com/final-form/react-final-form/releases/tag/v5.1.0).
